v2/bundle/spiffebundle/set.go

```diff
@@ -1,49 +1,97 @@
 package spiffebundle
 
 import (
+	"sync"
+
 	"github.com/spiffe/go-spiffe/v2/bundle/jwtbundle"
 	"github.com/spiffe/go-spiffe/v2/bundle/x509bundle"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
 )
 
 // Set is a set of bundles, keyed by trust domain.
-type Set struct{}
+type Set struct {
+	mtx     sync.RWMutex
+	bundles map[spiffeid.TrustDomain]*Bundle
+}
 
 // NewSet creates a new set initialized with the given bundles.
 func NewSet(bundles ...*Bundle) *Set {
-	panic("not implemented")
+	bundlesMap := make(map[spiffeid.TrustDomain]*Bundle)
+
+	for _, b := range bundles {
+		if b != nil {
+			bundlesMap[b.trustDomain] = b
+		}
+	}
+
+	return &Set{
+		bundles: bundlesMap,
+	}
 }
 
 // Add adds a new bundle into the set. If a bundle already exists for the
 // trust domain, the existing bundle is replaced.
 func (s *Set) Add(bundle *Bundle) {
-	panic("not implemented")
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if bundle != nil {
+		s.bundles[bundle.trustDomain] = bundle
+	}
 }
 
 // Remove removes the bundle for the given trust domain.
 func (s *Set) Remove(trustDomain spiffeid.TrustDomain) {
-	panic("not implemented")
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	delete(s.bundles, trustDomain)
 }
 
 // Has returns true if there is a bundle for the given trust domain.
 func (s *Set) Has(trustDomain spiffeid.TrustDomain) bool {
-	panic("not implemented")
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+
+	_, ok := s.bundles[trustDomain]
+	return ok
 }
 
 // GetBundleForTrustDomain returns the SPIFFE bundle for the given trust
 // domain. It implements the Source interface.
 func (s *Set) GetBundleForTrustDomain(trustDomain spiffeid.TrustDomain) (*Bundle, error) {
-	panic("not implemented")
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+
+	if !s.Has(trustDomain) {
```

Comment:
Hmm, i didn't pick this up in the other Set PRs, but we shouldn't use `Has()` in these methods, since it also takes a read lock. We should just do the map lookup directly, IMO, expecially since we need the bundle anyway and there is no reason to do the lookup twice.